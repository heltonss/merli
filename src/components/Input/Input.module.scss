@import '../../assets/scss/colors';

.container {
  width: 100%;
  height: 36px;
  padding-left: 16px;

  &::placeholder {
    color: $gray;
  }

  &::-ms-input-placeholder {
    color: $gray;
  }

  &::-ms-input-placeholder {
    color: $gray;
  }
}
