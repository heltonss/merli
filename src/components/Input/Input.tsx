import classnames from 'classnames';
import React from 'react';
import Style from './Input.module.scss';

interface IOwnProps {
  placeholder: string;
  type: string;
  parentClassStyle?: string;
  getValue(value: string): void;
}

const Input = ({ placeholder, type, getValue, parentClassStyle }: IOwnProps) => {
  const getValueInput = (event: any) => {
    getValue(event.target.value);
  };
  return (
    <input
      className={classnames([Style.container, parentClassStyle])}
      type={type}
      placeholder={placeholder}
      onChange={getValueInput}
    />
  );
};

export default Input;
