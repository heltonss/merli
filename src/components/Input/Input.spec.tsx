import { mount, shallow } from 'enzyme';
import React from 'react';
import Input from './Input';

describe('Input', () => {
  it('Should render without crash', () => {
    const wrapper = shallow(<Input placeholder={'insira seu texto aqui'} type={'text'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should get the props', () => {
    const getValueInput = () => {
      return;
    };
    const wrapper = mount(
      <Input
        placeholder={'Buscar produtos, marcas, muito mais'}
        type={'text'}
        getValue={getValueInput}
      />,
    );
    expect(wrapper.props().placeholder).toEqual('Buscar produtos, marcas, muito mais');
    expect(wrapper.props().type).toEqual('text');
  });
});
