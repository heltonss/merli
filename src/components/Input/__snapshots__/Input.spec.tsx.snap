// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render without crash 1`] = `ShallowWrapper {}`;
